modules/migration-understanding-cpma.adoc

```diff
@@ -0,0 +1,41 @@
+// Module included in the following assemblies:
+// migration/migrating-openshift-3-to-4.adoc
+[id='migration-understanding-cpma_{context}']
+= Understanding the Control Plane Migration Assistant
+
+The Control Plane Migration Assistant (CPMA) is a CLI-based tool that assists you in migrating the control plane from {product-title} 3.x to 4.x. The CPMA processes {product-title} 3.x configuration files and generates Custom Resource (CR) manifest files, which are consumed by {product-title} 4.x operators.
+
+Because {product-title} 3 and 4 have significant configuration differences, not all parameters are processed. CPMA generates a report that describes whether features are supported fully, partially, or not at all.
+
+.Configuration files
+
+CPMA uses the Kubernetes and {product-title} APIs to access the following configuration files on an {product-title} 3 cluster:
+
+* Master configuration file (default: *_/etc/origin/master/master-config.yaml_*)
+* CRI-O configuration file (default: *_/etc/crio/crio.conf_*)
+* etcd configuration file (default: *_/etc/etcd/etcd.conf_*)
+* Image registries file (default: *_/etc/containers/registries.conf_*)
+* Dependent configuration files:
+** Password files (for example, HTPasswd)
+** ConfigMaps
+** Secrets
+
+.CR Manifests
+
+CPMA generates CR manifests for the following configurations:
+
+* API server CA certificate: *_100_CPMA-cluster-config-APISecret.yaml_*
++
+[NOTE]
+====
+If you are using an unsigned certificate, you must link:https://docs.openshift.com/container-platform/4.1/authentication/certificates/api-server.html#add-named-api-server_api-server-certificates[add the API server certificate].
```

Comment:
You can't link here. We post the OpenShift docs both on docs.openshift.com and the customer portal, so this link would be incorrect on the portal. So unfortunately we just can't link at all to docs from modules right now.

Reply:
So how do you link to different parts of your documentation?

Reply:
Are relative links OK?

link:../openshift_images/image-streams-manage.html#images-imagestreams-update-tag_image-streams-managing[update the `imagestreamtags` references]

Reply:
Never mind -- I see that relative links are OK - https://github.com/openshift/openshift-docs/blob/dd2a3461de1e2ddc61b98c7bb3186e68950edd4e/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references

Reply:
You can't link to other parts of the docs when you're inside a module. You can only use xrefs as long as you're in an assembly.

Your link for the guidelines is outdated. See: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#links-hyperlinks-and-cross-references